### PR TITLE
feat: traversal paths

### PR DESCRIPTION
**Pull request by cristianmtr**
add traversal paths

### CODE REVIEW

File: executor.py [hunk 1]
```diff
@@ -76,8 +76,10 @@ def search(
             else self._match_args
         )
 
+        traversal_right = parameters.get('traversal_right', '@r')
+        traversal_left = parameters.get('traversal_left', '@r')
         match_args = SimpleIndexer._filter_match_params(docs, match_args)
-        docs.match(self._index, **match_args)
+        docs[traversal_left].match(self._index[traversal_right], **match_args)
```
Comment (samsja) [inline executor.py:82]: The old way of doing it it to pass by match_args : https://github.com/jina-ai/executor-simpleindexer/tree/v0.11 It is common to a lot of our executor.

The way is the code now is that is merge match_args with the parameters so it should be 
```python
traversal_right = match_args.get('traversal_right', '@r')
traversal_left = match_args.get('traversal_left', '@r')
```

**Review by samsja — DISMISSED** [review]

File: executor.py (hunk 1 above)
Comment (cristianmtr) [reply executor.py:82]: But it's not a thing anymore in `match()`. See [here](https://docarray.jina.ai/api/docarray.array.mixins.match/#docarray.array.mixins.match.MatchMixin.match) So it shouln't be part of match_args

File: executor.py [hunk 2]
```diff
@@ -76,8 +76,10 @@ def search(
             else self._match_args
         )
 
+        traversal_right = parameters.get('traversal_right', '@r')
```
Comment (JoanFM) [inline executor.py:79]: can we have default ones at `__init__` time?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: executor.py (hunk 2 above)
Comment (cristianmtr) [reply executor.py]: yes. re-check pls

**Review by JoanFM — APPROVED** [review]

File: executor.py (hunk 1 above)
Comment (samsja) [reply executor.py]: It was more to keep old code working but you are right then

**Pull request merged**